lib/matplotlib/backends/backend_qtagg.py

```diff
@@ -47,25 +46,20 @@ def paintEvent(self, event):
             right = left + width
             # create a buffer using the image bounding box
             bbox = Bbox([[left, bottom], [right, top]])
-            buf = memoryview(self.copy_from_bbox(bbox))
+            region = self.copy_from_bbox(bbox)
 
-            if QT_API == "PyQt6":
-                from PyQt6 import sip
-                ptr = int(sip.voidptr(buf))
-            else:
-                ptr = buf
+            # Now draw the region we copied with the painter.
+            r_width, r_height = region.get_bounds()[2:]
 
-            painter.eraseRect(rect)  # clear the widget canvas
-            qimage = QtGui.QImage(ptr, buf.shape[1], buf.shape[0],
-                                  QtGui.QImage.Format.Format_RGBA8888)
+            # Draw the region copied with the QPainter
+            qimage = QtGui.QImage(
+                np.frombuffer(region, dtype=np.uint8),
+                r_width, r_height,
+                QtGui.QImage.Format.Format_RGBA8888,
+            )
             qimage.setDevicePixelRatio(self.device_pixel_ratio)
             # set origin using original QT coordinates
-            origin = QtCore.QPoint(rect.left(), rect.top())
-            painter.drawImage(origin, qimage)
-            # Adjust the buf reference count to work around a memory
-            # leak bug in QImage under PySide.
-            if QT_API == "PySide2" and QtCore.__version_info__ < (5, 12):
```

Comment:
Looking at #25363 I guess this branch is never exercised anymore indeed, but the qt>=5.12 version bound doesn't seem to be documented anywhere but in that PR changelog note; perhaps it should be mentioned in dependencies.rst (even though it is already a consequence of wheel availability for py3.9)?  (@oscargus)